Tools/MiniBrowser/haiku/App.cpp

```diff
@@ -10,26 +10,103 @@
 
 #include "App.h"
 
+#include <interface/StringView.h>
+#include <Button.h>
+#include <Entry.h>
+#include <GridLayoutBuilder.h>
+#include <GroupLayout.h>
+#include <GroupLayoutBuilder.h>
+#include <MenuBar.h>
+#include <MenuItem.h>
+#include <SeparatorView.h>
+#include <SpaceLayoutItem.h>
+#include <StatusBar.h>
+#include <StringView.h>
+#include <TextControl.h>
 
+#include "WebViewConstants.h"
+enum {
+	OPEN_LOCATION = 'open',
+	OPEN_INSPECTOR = 'insp',
+	SAVE_PAGE = 'save',
+    GO_BACK = 'goba',
+    GO_FORWARD = 'gofo',
+    STOP = 'stop',
+    GOTO_URL = 'goul',
+    RELOAD = 'reld',
 
-enum
-{
-  M_BUTTON_GO_URL = 'urlg'
+    TEXT_SIZE_INCREASE = 'tsin',
+    TEXT_SIZE_DECREASE = 'tsdc',
+    TEXT_SIZE_RESET = 'tsrs',
 };
 
 App::App(void)
     :    BApplication("application/x-vnd.lh-MiniBrowser")
 {
-  //initialize counter
-  fCount = 0;
-
-  //create window
-  frame.Set(100,100,500,400);
-  myWindow = new BWindow(frame,"MiniBrowser"
-                                 , B_TITLED_WINDOW
-                                 , B_ASYNCHRONOUS_CONTROLS
-                                   | B_QUIT_ON_WINDOW_CLOSE);
-  myWindow->Show();
+	//initialize counter
+	fCount = 0;
+	
+	//create window
+	frame.Set(100,100,800,800);
+	myWindow = new BWindow(frame,"MiniBrowser"
+	                               , B_DOCUMENT_WINDOW_LOOK,B_NORMAL_WINDOW_FEEL
+	                               , B_AUTO_UPDATE_SIZE_LIMITS | B_ASYNCHRONOUS_CONTROLS
+	                                 | B_QUIT_ON_WINDOW_CLOSE );
+	                                 
+	myWindow->SetLayout(new BGroupLayout(B_HORIZONTAL));
+	
+	                                 
+	m_menuBar = new BMenuBar("Main menu");
+	BMenu* menu = new BMenu("Window");
+	BMessage* newWindowMessage = new BMessage(NEW_WINDOW);
+	newWindowMessage->AddString("url", "");
+	BMenuItem* newItem = new BMenuItem("New", newWindowMessage, 'N');
+	menu->AddItem(newItem);
+	newItem->SetTarget(be_app);
+	menu->AddItem(new BMenuItem("Open location", new BMessage(OPEN_LOCATION), 'L'));
+	menu->AddItem(new BMenuItem("Inspect page", new BMessage(OPEN_INSPECTOR), 'I'));
+	menu->AddItem(new BMenuItem("Save page", new BMessage(SAVE_PAGE), 'S'));
+	menu->AddSeparatorItem();
+	menu->AddItem(new BMenuItem("Close", new BMessage(B_QUIT_REQUESTED), 'W', B_SHIFT_KEY));
+	BMenuItem* quitItem = new BMenuItem("Quit", new BMessage(B_QUIT_REQUESTED), 'Q');
+	menu->AddItem(quitItem);
+	quitItem->SetTarget(be_app);
+	m_menuBar->AddItem(menu);
+	
+	menu = new BMenu("Text");
+	menu->AddItem(new BMenuItem("Increase size", new BMessage(TEXT_SIZE_INCREASE), '+'));
+	menu->AddItem(new BMenuItem("Decrease size", new BMessage(TEXT_SIZE_DECREASE), '-'));
+	menu->AddItem(new BMenuItem("Reset size", new BMessage(TEXT_SIZE_RESET), '0'));
+	m_menuBar->AddItem(menu);
+	
+	
+	m_BackButton = new BButton("","Back",new BMessage(GO_BACK));
+    m_ForwardButton = new BButton("","Forward", new BMessage(GO_FORWARD));
```

Comment:
what happened to the indentation here? We've been a bit inconsistent with it but at least try to use one type (either space or tabs) per function...